[Bug] Live Unit Testing Fails Build

<!-- Bug report best practices: https://github.com/xamarin/Xamarin.Forms/wiki/Submitting-Issues -->

### Description
Having a Xamarin Forms project in a solution with Live Unit Testing enabled destabilizes the unit testing build & runner. 
```
[10:51:37.518 Verbose] - BuildManager - D:\Users\~~~\.nuget\packages\xamarin.forms\4.2.0.815419\build\Xamarin.Forms.targets(107,3): error MSB4018: The "XamlCTask" task failed unexpectedly.
System.IO.FileNotFoundException: No symbol found for file: D:\Users\~~~\.vs\~~~\v16\lut\159548\i\~~~\Debug\netstandard2.0\~~~.Forms.dll
   at Mono.Cecil.Cil.DefaultSymbolReaderProvider.GetSymbolReader(ModuleDefinition module, String fileName)
   at Mono.Cecil.ModuleReader.ReadSymbols(ModuleDefinition module, ReaderParameters parameters)
   at Mono.Cecil.ModuleReader.CreateModule(Image image, ReaderParameters parameters)
   at Mono.Cecil.ModuleDefinition.ReadModule(String fileName, ReaderParameters parameters)
   at Mono.Cecil.AssemblyDefinition.ReadAssembly(String fileName, ReaderParameters parameters)
   at Xamarin.Forms.Build.Tasks.XamlCTask.Execute(IList`1& thrownExceptions)
   at Xamarin.Forms.Build.Tasks.XamlTask.Execute()
   at Microsoft.Build.Framework.ITask.Execute()
   at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
   at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext()
```

My experience was that two things would happen, even if the Forms project is not needed for tests:

1. Obviously, the build fails. LUT would tend to not report status of changes.
2. A file handle would be left open on libraries, so even with successful builds the LUT runner can't continue. (This required restarting Visual Studio to resolve, but would fail again soon)

### Reproduction
I can put together a reproduction project, although I suspect the failures are highly dependent upon system configuration and performance. I'd like to propose a solution that is a benefit to everyone, even if they don't see this behavior.

### Proposal
Unless there is a reason I'm not aware of, I'd suggest disabling the XamlC task for the LUT build.

```xml
<Target 
    Name="XamlC" 
    AfterTargets="AfterCompile"
    Inputs="$(IntermediateOutputPath)$(TargetFileName)"
    Outputs="$(IntermediateOutputPath)XamlC.stamp"
    Condition=" '$(DesignTimeBuild)' != 'True' And '$(BuildingForLiveUnitTesting)' != 'True' "
    >
```

### Workaround
In the meantime, I've added this to my project file:
```xml
<DesignTimeBuild Condition=" '$(BuildingForLiveUnitTesting)' == 'True' ">True</DesignTimeBuild>
```

--- a/.nuspec/Xamarin.Forms.targets
+++ b/.nuspec/Xamarin.Forms.targets
@@ -106,7 +106,7 @@
 		</CompileDependsOn>
 	</PropertyGroup>
 
-	<Target Name="XamlC" AfterTargets="AfterCompile" Inputs="$(IntermediateOutputPath)$(TargetFileName)" Outputs="$(IntermediateOutputPath)XamlC.stamp" Condition=" '$(DesignTimeBuild)' != 'True' ">
+	<Target Name="XamlC" AfterTargets="AfterCompile" Inputs="$(IntermediateOutputPath)$(TargetFileName)" Outputs="$(IntermediateOutputPath)XamlC.stamp" Condition=" '$(DesignTimeBuild)' != 'True' And '$(BuildingForLiveUnitTesting)' != 'True' ">
 		<XamlCTask
 			Assembly = "$(IntermediateOutputPath)$(TargetFileName)"
 			ReferencePath = "@(ReferencePath)"

--- a/.nuspec/Xamarin.Forms.targets
+++ b/.nuspec/Xamarin.Forms.targets
@@ -109,7 +109,7 @@
 	<Target Name="XamlC" AfterTargets="AfterCompile" Inputs="$(IntermediateOutputPath)$(TargetFileName)" Outputs="$(IntermediateOutputPath)XamlC.stamp" Condition=" '$(DesignTimeBuild)' != 'True'">
     <PropertyGroup>
       <ValidateOnlyValue>$(XFXamlCValidateOnly)</ValidateOnlyValue>
-      <ValidateOnlyValue Condition="'$(BuildingForLiveUnitTesting)' != 'True' ">True</ValidateOnlyValue>
+      <ValidateOnlyValue Condition="'$(BuildingForLiveUnitTesting)' == 'True' ">True</ValidateOnlyValue>
     </PropertyGroup>
 		<XamlCTask
 			Assembly = "$(IntermediateOutputPath)$(TargetFileName)"

--- a/.nuspec/Xamarin.Forms.targets
+++ b/.nuspec/Xamarin.Forms.targets
@@ -107,17 +107,17 @@
 	</PropertyGroup>
 
 	<Target Name="XamlC" AfterTargets="AfterCompile" Inputs="$(IntermediateOutputPath)$(TargetFileName)" Outputs="$(IntermediateOutputPath)XamlC.stamp" Condition=" '$(DesignTimeBuild)' != 'True'">
-    <PropertyGroup>
-      <ValidateOnlyValue>$(XFXamlCValidateOnly)</ValidateOnlyValue>
-      <ValidateOnlyValue Condition="'$(BuildingForLiveUnitTesting)' == 'True' ">True</ValidateOnlyValue>
-    </PropertyGroup>
+	    <PropertyGroup>
+		<_XFXamlCValidateOnly>$(XFXamlCValidateOnly)</_XFXamlCValidateOnly>
+		<_XFXamlCValidateOnly Condition="'$(BuildingForLiveUnitTesting)' == 'True' ">True</_XFXamlCValidateOnly>
+	    </PropertyGroup>
 		<XamlCTask
 			Assembly = "$(IntermediateOutputPath)$(TargetFileName)"
 			ReferencePath = "@(ReferencePath)"
 			OptimizeIL = "true"
 			DebugSymbols = "$(DebugSymbols)"
 			DebugType = "$(DebugType)"
-			ValidateOnly = "$(ValidateOnlyValue)"
+			ValidateOnly = "$(_XFXamlCValidateOnly)"
 			KeepXamlResources = "$(XFKeepXamlResources)" />
 		<Touch Files="$(IntermediateOutputPath)XamlC.stamp" AlwaysCreate="True" />
 		<ItemGroup>

--- a/.nuspec/Xamarin.Forms.targets
+++ b/.nuspec/Xamarin.Forms.targets
@@ -106,14 +106,18 @@
 		</CompileDependsOn>
 	</PropertyGroup>
 
-	<Target Name="XamlC" AfterTargets="AfterCompile" Inputs="$(IntermediateOutputPath)$(TargetFileName)" Outputs="$(IntermediateOutputPath)XamlC.stamp" Condition=" '$(DesignTimeBuild)' != 'True' And '$(BuildingForLiveUnitTesting)' != 'True' ">
+	<Target Name="XamlC" AfterTargets="AfterCompile" Inputs="$(IntermediateOutputPath)$(TargetFileName)" Outputs="$(IntermediateOutputPath)XamlC.stamp" Condition=" '$(DesignTimeBuild)' != 'True'">
+    <PropertyGroup>
+      <ValidateOnlyValue>$(XFXamlCValidateOnly)</ValidateOnlyValue>
+      <ValidateOnlyValue Condition="'$(BuildingForLiveUnitTesting)' != 'True' ">True</ValidateOnlyValue>
+    </PropertyGroup>
 		<XamlCTask
 			Assembly = "$(IntermediateOutputPath)$(TargetFileName)"
 			ReferencePath = "@(ReferencePath)"
 			OptimizeIL = "true"
 			DebugSymbols = "$(DebugSymbols)"
 			DebugType = "$(DebugType)"
-			ValidateOnly = "$(XFXamlCValidateOnly)"
+			ValidateOnly = "$(ValidateOnlyValue)"
 			KeepXamlResources = "$(XFKeepXamlResources)" />
 		<Touch Files="$(IntermediateOutputPath)XamlC.stamp" AlwaysCreate="True" />
 		<ItemGroup>

